src/views/ReviewConfirm.vue

```diff
@@ -6,18 +6,70 @@
 
     <v-card flat class="review-confirm-container">
       <section>
-        Review/confirm component goes here.
+        <!--Summary component goes here -->
+      </section>
+      <section>
+          <header>
+                <h2>2. Completing Party Statement</h2>
+              </header>
+              <Certify
+                :isCertified.sync="isCertified"
+                :certifiedBy.sync="certifiedBy"
+                @valid="certifyFormValid=$event"
+                :date="currentDate"
+                :certifyStatementResource="certifyStatementResource"/>
       </section>
     </v-card>
   </div>
 </template>
 
 <script lang="ts">
 // Libraries
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Mixins, Watch } from 'vue-property-decorator'
+import { State, Action } from 'vuex-class'
+
+import { Certify } from '@/components/ReviewConfirm'
+import { ResourceLookupMixin } from '../mixins'
+import { CertifyStatementIF, ActionBindingIF } from '../interfaces'
+
+@Component({
+  components: {
+    Certify
+  }
+})
+export default class ReviewConfirm extends Mixins(ResourceLookupMixin) {
+   @State(state => state.stateModel.currentDate) readonly currentDate!: string
+
+   @State(state => state.resourceModel.certifyStatementResource)
+   readonly certifyStatementResource!: CertifyStatementIF
 
-@Component({})
-export default class ReviewConfirm extends Vue {}
+   @Action('setCertifyState') setCertifyState!: ActionBindingIF
+
+  // properties for Certify component
+  private certifiedBy:string = ''
+  private isCertified:boolean = false
+  private certifyFormValid:boolean = false
+
+  @Watch('certifyFormValid')
+  private onCertifyFormValid (val: boolean): void {
+    this.setCertifyState(
+      {
+        certifyFormValid: val,
+        certifiedBy: this.certifiedBy
+      }
+    )
+  }
+
+  @Watch('certifiedBy')
+  private onCertifiedByChange (val: boolean): void {
+    this.setCertifyState(
```

Comment:
This code is overly complicated as it has too much going on.

If I understand correctly, when the user enters "certified by", the certify component emits an event that updates the local property certifiedBy, which is watched in order to set the global certify state.

I think this should be simplified! I think you could remove the "sync" from the props and then call a handler method for each prop update event. The handlers could call setCertifyState(). This would basically replace the watchers with event handlers but it would cut out one interaction (ie, call handler directly instead of watch) and I think it would be much easier to understand.

Or maybe there's another way to bind the props to the states?

What do you think?

Reply:
I dont think it is complicated and we have similar interaction patterns in coops. But in this scenario I agree that use of a watcher and thus one interaction can be avoided. Will make the change to invoke a handler.

Reply:
The component is modified to use handler. As part of this I removed the prop isCertify which does not make any sense. The parent component need not pass that to certify even in the case of save and resume

Reply:
Thank you for the handler usage. It makes the code easier for me to understand and review :)